### ISSUE TITLE
Wrong image for azul/zulu-openjdk-alpine:8 pushed

### ISSUE THREAD

**Issue by henrist** [zulu-openjdk/zulu-openjdk#63]
I have a job that updates our images every night, which pulls the latest `azul/zulu-openjdk-alpine:8`.

It seems this was resolved to a JRE build and not a JDK build, causing downstreams jobs to fail. The tag appers to later have been corrected and now properly resolves to a JDK build.

What happened between 34 and 14 hours ago? Can this happen again so that the image contents cannot be trusted to reflect the correct Dockerfile?

## Details

From our build log (that ran 2020-05-08 04:35 UTC) I can see this:

```
Step 1/2 : FROM azul/zulu-openjdk-alpine:8
8: Pulling from azul/zulu-openjdk-alpine
cbdbe7a5bc2a: Already exists
2c23def1dd80: Pulling fs layer
2c23def1dd80: Download complete
2c23def1dd80: Pull complete
Digest: sha256:cae5887524ad90564eb63ca8af558d759901eb4048768179cd8ba2ab3004b479
Status: Downloaded newer image for azul/zulu-openjdk-alpine:8
```

This digest corresponds to the current tags `jre-headless`, `8-jre-headless`, `8u252-jre-headless` and more. From the history of it:

```
docker history --no-trunc azul/zulu-openjdk-alpine:8@sha256:cae5887524ad90564eb63ca8af558d759901eb4048768179cd8ba2ab3004b479
IMAGE                                                                     CREATED             CREATED BY                                                                                                                                                                                                                          SIZE                COMMENT
sha256:2eda09d46d5cfd9011f92792edbabfa0aadf063724ee0d4a4b1b212856889327   34 hours ago        /bin/sh -c #(nop)  ENV JAVA_HOME=/usr/lib/jvm/zulu8-ca                                                                                                                                                                              0B
<missing>                                                                 34 hours ago        /bin/sh -c wget --quiet https://cdn.azul.com/public_keys/alpine-signing@azul.com-5d5dc44c.rsa.pub -P /etc/apk/keys/ &&     echo "https://repos.azul.com/zulu/alpine" >> /etc/apk/repositories &&     apk --no-cache add zulu8-jre   110MB
<missing>                                                                 34 hours ago        /bin/sh -c #(nop)  ENV LC_ALL=en_US.UTF-8                                                                                                                                                                                           0B
<missing>                                                                 34 hours ago        /bin/sh -c #(nop)  ENV LANGUAGE=en_US:en                                                                                                                                                                                            0B
<missing>                                                                 34 hours ago        /bin/sh -c #(nop)  ENV LANG=en_US.UTF-8                                                                                                                                                                                             0B
<missing>                                                                 2 weeks ago         /bin/sh -c #(nop)  CMD ["/bin/sh"]                                                                                                                                                                                                  0B
<missing>                                                                 2 weeks ago         /bin/sh -c #(nop) ADD file:b91adb67b670d3a6ff9463e48b7def903ed516be66fc4282d22c53e41512be49 in /                                                                                                                                    5.61MB
```

As you can see, this digest contains the JRE and not the JDK.

According to [Docker Hub](https://hub.docker.com/layers/azul/zulu-openjdk-alpine/8/images/sha256-70284b2c42d349bc56f34bb88e3025ec4e5087c61694df766c636eb96fe5ddb0) the current tag was pushed 14 hours ago. That should be around 2020-05-08 07:xx UTC. This resolves to the digest `70284b2c42d349bc56f34bb88e3025ec4e5087c61694df766c636eb96fe5ddb0`.

The history of the _current_ image:

```
docker history --no-trunc azul/zulu-openjdk-alpine:8@sha256:70284b2c42d349bc56f34bb88e3025ec4e5087c61694df766c636eb96fe5ddb0
IMAGE                                                                     CREATED             CREATED BY                                                                                                                                                                                                                          SIZE                COMMENT
sha256:a814cad266418474a3e5a62f4d5756007d1592bda323d526037ce02c79c9ab27   14 hours ago        /bin/sh -c #(nop)  ENV JAVA_HOME=/usr/lib/jvm/zulu8-ca                                                                                                                                                                              0B
<missing>                                                                 14 hours ago        /bin/sh -c wget --quiet https://cdn.azul.com/public_keys/alpine-signing@azul.com-5d5dc44c.rsa.pub -P /etc/apk/keys/ &&     echo "https://repos.azul.com/zulu/alpine" >> /etc/apk/repositories &&     apk --no-cache add zulu8-jdk   150MB
<missing>                                                                 14 hours ago        /bin/sh -c #(nop)  ENV LC_ALL=en_US.UTF-8                                                                                                                                                                                           0B
<missing>                                                                 14 hours ago        /bin/sh -c #(nop)  ENV LANGUAGE=en_US:en                                                                                                                                                                                            0B
<missing>                                                                 14 hours ago        /bin/sh -c #(nop)  ENV LANG=en_US.UTF-8                                                                                                                                                                                             0B
<missing>                                                                 2 weeks ago         /bin/sh -c #(nop)  CMD ["/bin/sh"]                                                                                                                                                                                                  0B
<missing>                                                                 2 weeks ago         /bin/sh -c #(nop) ADD file:b91adb67b670d3a6ff9463e48b7def903ed516be66fc4282d22c53e41512be49 in /                                                                                                                                    5.61MB
```

**Comment by henrist** [comment]
I realize I didn't check for closed issues and that this duplicates #61. However it would be great to hear what happened and if this might be issues in the future as well.

**Comment by sgrinev** [comment]
Thanks for the reporting, it seems there was a conflict between new deployment and testing infra. Fixing...
